app/controllers/session_controller.rb

```diff
@@ -0,0 +1,45 @@
+
+class SessionController < ApplicationController
+
+  def authenticate
+    config = OpenIDConnect::Discovery::Provider::Config.discover! "https://did.app"
+
+    client = OpenIDConnect::Client.new({
+      identifier: ENV["CLIENT_ID"],
+      secret: ENV["CLIENT_SECRET"],
+      redirect_uri: "http://localhost:3000/session/callback",
+      issuer: config.issuer,
+      authorization_endpoint: config.authorization_endpoint,
+      jwks_uri: config.jwks_uri,
+      token_endpoint: config.token_endpoint,
+    })
+    redirect_to client.authorization_uri()
+  end
+
+  def callback
+    config = OpenIDConnect::Discovery::Provider::Config.discover! "https://did.app"
+
+    client = OpenIDConnect::Client.new({
+      identifier: ENV["CLIENT_ID"],
```

Comment:
Is calling straight into ENV the right thing to do here?

Reply:
Yes!